universal-application-tool-0.0.1/app/views/AzureFileUploadViewStrategy.java

```diff
@@ -0,0 +1,142 @@
+        .with(input().withType("hidden").withName("fileName").withValue(request.fileName()))
+        .with(input().withType("hidden").withName("accountName").withValue(request.accountName()))
+        .with(
+            input()
+                .withType("hidden")
+                .withName("successActionRedirect")
+                .withValue(request.successActionRedirect()))
+        .with(errorDiv(params.messages(), fileUploadQuestion));
+  }
+
+  @Override
+  public Tag renderFileUploadBlockSubmitForms(
+      Params params, ApplicantQuestionRendererFactory applicantQuestionRendererFactory) {
+    String key = FileNameFormatter.formatFileUploadQuestionFilename(params);
+
+    String onSuccessRedirectUrl =
+        params.baseUrl()
+            + routes.ApplicantProgramBlocksController.updateFile(
+                    params.applicantId(),
+                    params.programId(),
+                    params.block().getId(),
+                    params.inReview())
+                .url();
+
+    StorageUploadRequest request =
+        params.storageClient().getSignedUploadRequest(key, onSuccessRedirectUrl);
+
+    BlobStorageUploadRequest blobStorageUploadRequest = castStorageRequest(request);
+
+    ApplicantQuestionRendererParams rendererParams =
+        ApplicantQuestionRendererParams.builder()
+            .setMessages(params.messages())
+            .setSignedFileUploadRequest(blobStorageUploadRequest)
+            .build();
+
+    ContainerTag formTag =
+        form()
+            .withId(BLOCK_FORM_ID)
+            .with(
+                each(
+                    params.block().getQuestions(),
+                    question ->
+                        renderQuestion(
+                            question, rendererParams, applicantQuestionRendererFactory)));
+
+    Tag buttons = renderFileUploadBottomNavButtons(params);
+
+    return div(formTag, buttons)
+        .withId("azure-upload-form-component")
+        .with(
+            footer(viewUtils.makeWebJarsTag(WebJarJsPaths.AZURE_STORAGE_BLOB.getString())),
+            footer(viewUtils.makeLocalJsTag("azure_upload")));
+  }
+
+  private BlobStorageUploadRequest castStorageRequest(StorageUploadRequest request) {
+    if (!(request instanceof BlobStorageUploadRequest)) {
+      throw new RuntimeException(
+          "Tried to upload a file to Azure Blob storage using incorrect request type");
+    }
+    return (BlobStorageUploadRequest) request;
+  }
+
+  private Tag renderFileUploadBottomNavButtons(Params params) {
+    ContainerTag ret =
+        div()
+            .withClasses(ApplicantStyles.APPLICATION_NAV_BAR)
+            // An empty div to take up the space to the left of the buttons.
+            .with(div().withClasses(Styles.FLEX_GROW))
+            .with(renderReviewButton(params));
+
+    ret.with(renderAzureUploadButton(params));
+
+    return ret;
+  }
+
+  private ContainerTag renderAzureUploadButton(Params params) {
```

Comment:
i feel like this is really similar to the aws one, did you consider pushing this up to the parent class as well?

Reply:
Done